conn_linux_integration_test.go

```diff
@@ -0,0 +1,67 @@
+//+build integration,linux
+
+package netlink
+
+import (
+	"fmt"
+	"os/exec"
+	"testing"
+	"time"
+)
+
+func TestLinuxNetlinkMulticast(t *testing.T) {
+	cfg := &Config{
+		Groups: 0x1, // RTMGRP_LINK
+	}
+
+	c, err := Dial(0, cfg) // dials NETLINK_ROUTE
+	if err != nil {
+		t.Fatalf("failed to dial: %v", err)
+
+	}
+
+	in := make(chan []Message)
+
+	// routine for receiving any messages
+	recv := func() {
+		data, er := c.Receive()
+		if er != nil {
+			fmt.Printf("error in receive: %s\n", err)
+		}
+		in <- data
+	}
+
+	go recv()
+
+	def := sudoIfCreate(t, "test0")
+	defer def()
+
+	timeout := time.After(5 * time.Second)
+	var got []Message
+	select {
+	case got = <-in:
+		break
+	case <-timeout:
+		t.Fatal("did not receive any messages after 5 seconds\n")
+		break
+	}
+	t.Logf("received netlink data: %#v\n", got)
```

Comment:
Obviously this isn't a `rtnetlink` package, but are there any kind of header values we can inspect quickly here just to determine that this is a valid response and not something bogus?